src/integrationTest/kotlin/org/radarbase/gateway/resource/KafkaTopicsTest.kt

```diff
@@ -90,217 +90,246 @@ class KafkaTopicsTest {
-        results += sendData(BASE_URI, retriever, topic, accessToken, key, value, binary = true, gzip = false)
-        results += sendData(BASE_URI, retriever, topic, accessToken, key, value, binary = false, gzip = true)
-        results += sendData(BASE_URI, retriever, topic, accessToken, key, value, binary = false, gzip = false)
+        gatewayContext.sendData(binary = true, gzip = true)
+        results += gatewayContext.sendData(
+            binary = true,
+            gzip = true,
+        )
+        results += gatewayContext.sendData(
+            binary = true,
+            gzip = false,
+        )
+        results += gatewayContext.sendData(
+            binary = false,
+            gzip = true,
+        )
+        results += gatewayContext.sendData(
+            binary = false,
+            gzip = false,
+        )
         results.forEach { println(it) }
 
-        httpClient.call(Status.OK) {
-            url("$BASE_URI/topics")
-            head()
-            addHeader("Authorization", BEARER + accessToken)
+        httpClient.head("$BASE_URI/topics") {
+            bearer(gatewayContext.accessToken)
+        }.let { response ->
+            assertThat(response.status, equalTo(HttpStatusCode.OK))
         }
 
-        httpClient.call(Status.UNAUTHORIZED) {
-            url("$BASE_URI/topics")
+        httpClient.get("$BASE_URI/topics").let { response ->
+            assertThat(response.status, equalTo(HttpStatusCode.Unauthorized))
         }
 
-        httpClient.call(Status.UNAUTHORIZED) {
-            url("$BASE_URI/topics").head()
+        httpClient.get("$BASE_URI/topics").let { response ->
+            assertThat(response.status, equalTo(HttpStatusCode.Unauthorized))
         }
 
-        httpClient.call(422) {
-            url("$BASE_URI/topics/test")
-            post("{}".toRequestBody(JSON_TYPE))
-            addHeader("Authorization", BEARER + accessToken)
+        httpClient.head("$BASE_URI/topics").let { response ->
+            assertThat(response.status, equalTo(HttpStatusCode.Unauthorized))
         }
 
-        httpClient.call(422) {
-            url("$BASE_URI/topics/test")
-            post("{}".toRequestBody(KAFKA_JSON_TYPE))
-            addHeader("Authorization", BEARER + accessToken)
+        httpClient.post("$BASE_URI/topics/test") {
+            bearer(gatewayContext.accessToken)
+            setBody("{}")
+            contentType(ContentType.Application.Json)
+        }.let { response ->
+            assertThat(response.status, equalTo(HttpStatusCode.UnprocessableEntity))
         }
 
-        httpClient.call(Status.BAD_REQUEST) {
-            url("$BASE_URI/topics/test")
-            post("".toRequestBody(KAFKA_JSON_TYPE))
-            addHeader("Authorization", BEARER + accessToken)
+        httpClient.post("$BASE_URI/topics/test") {
+            bearer(gatewayContext.accessToken)
+            setBody("{}")
```

Comment:
this is a repeat of the last test if i'm not mistaken

```suggestion
            setBody("")
```